### DISCUSSION TITLE
How to set RTX Super Resolution to match display resolution?

### DISCUSSION THREAD

**Discussion by ghost** [mpv-player/mpv#14804] [Q&A] [answered]
Hi, I set the config file so that mpv uses RTX Super Resolution to perform upscaling
```
vo=gpu-next
hwdec=d3d11va
gpu-api=d3d11
scale=ewa_lanczos4sharpest
vf=d3d11vpp=scale=2:scaling-mode=nvidia
```
Everything works fine. But is there a way to set scale={value} to automatically change depending on the starting resolution of the video file and the display that is being used?
So that I would get
```
scale=1             # for 2560x1440 > 2560x1440 and for 3840x2160 > 3840x2160
scale=1.3333        # for 1920x1080 > 2560x1440
scale=2             # for 1280x720 > 2560x1440 and for 1920x1080 > 3840x2160
scale=3             # for 1280x720 > 3840x2160
etc...
```

**Comment by Andarwinux** [comment, marked as answer]
```
function autovsr()
    display_width = mp.get_property_native("display-width")
    video_width = mp.get_property_native("width")
    display_height = mp.get_property_native("display-height")
    video_height = mp.get_property_native("height")
    pixfmt = mp.get_property_native("video-params/hw-pixelformat") or mp.get_property_native("video-params/pixelformat")
    if video_width ~= nil and display_width ~= nil then 
        scale = math.max(display_width,display_height) / math.max(video_width,video_height)
        scale = scale - scale % 0.1
        if string.match(mp.get_property("vf"),"@vsr") then
            mp.command("vf remove @vsr")
        end
        if scale > 1 then
            if pixfmt == "nv12" or pixfmt == "yuv420p" then
                mp.command("vf append @vsr:d3d11vpp:scaling-mode=nvidia:scale="..scale)
            end
        end
    end
end
mp.observe_property("video-params/pixelformat", "native", autovsr)
mp.observe_property("vf", "native", autovsr)
```

**Reply by ghost** [reply]
Oh wow, this is perfect, thank you so much!

**Reply by prochak96** [reply]
Hi, so how does one actually get this function running in mpv? Do I need to put this in a Lua script in the scripts folder?

**Reply by KiNGRiCHII** [reply]
Ok, can someone please just explain to me if it is possible, to get RTX Super Resolution (setting 4) to Always upscale the quality of ANY VIDEO, No matter what the starting Resolution is, in Edge Browser when streaming video, and to upscale whether or not it is in fullscreen, so that even thumbnails, or any still image, or anything that is encoded with a video or picture file type gets upscaled x4 via RTX Super Resolution?

**Reply by azumukupoe** [reply]
@Andarwinux 
I tweaked the script slightly to ensure proper toggling. I’ve uploaded it as a Gist here: https://gist.github.com/azumukupoe/a56a78d71cf26a8f682dc06407e615db. Appreciate your original idea!

**Comment by Snuffpig** [comment]
Now, how to get this into mpv upstream so that scaling behaviour is handled like this gracefully by default?

**Reply by datrumole** [reply]
> Now, how to get this into mpv upstream so that scaling behaviour is handled like this gracefully by default?

an additional option is you can also do this in a conditional profile, but agreed, having it auto-scale in the code would be preferred 

below is based on a 4k output, adjust 'scale=x' accordingly 

[4k]
profile-desc=4k
profile-cond=(width ==3840)
vf=d3d11vpp=scale=1:scaling-mode=nvidia

[full-hd]
profile-desc=full-hd
profile-cond=(width ==1920)
vf=d3d11vpp=scale=2:scaling-mode=nvidia

[hd]
profile-desc=hd
profile-cond=(width ==1280)
vf=d3d11vpp=scale=3:scaling-mode=nvidia

**Reply by KiNGRiCHII** [reply]
wow awesome... And where is this file to be placed EXACTLY.
And thanks a ton for your work. I look forward to testing the new version with graceful handling, because I have been having issues with it activating too strongly at my native 2-k res.. somuchso that I've had to avoid playing anything at that res... if it's even working though, or if it is some other tweak out of the thousands I've seemingly applied to my PC and registry, I don't know... Because the file is currently placed in the Root of the edge browser folder... past the 1st root that is... where the actual browser .exe in use is.
 ...[update/edit] my bad reading on, I see that I probably never ustilized it correctly at all. but I'm looking for live streaming browser compatible solutions... however, it is interesting enough to make me wander where else could this file be utilized??

**Reply by datrumole** [reply]
the conditional profiles above would be part of the mpv.conf file

you would need to adjust the scaling based on your monitor, the above assumes a 4k display 

scale=1                 # for 2560x1440 -> 2560x1440 and for 3840x2160 -> 3840x2160
scale=1.3333        # for 1920x1080 -> 2560x1440
scale=2                 # for 1280x720 -> 2560x1440 and for 1920x1080 -> 3840x2160
scale=3                 # for 1280x720 -> 3840x2160

**Comment by Snuffpig** [comment]
[autovsr.txt](https://github.com/user-attachments/files/19197488/autovsr.txt)

This also might help (rename it .lua)

**Comment by drandarov-io** [comment]
Does this still work with the newest drivers. I tried all the attached scripts above and I even see the toggle text, but NVIDA App tells me that Super Resolution is not active.

**Reply by drandarov-io** [reply]
For anyone in the future. This requires at least 0.39 and Plex for example ships with 0.38. So replacing the `libmpv-2.dll` in `C:\Program Files\Plex\Plex` with an up to date for for example from https://github.com/zhongfly/mpv-winbuild/releases (note the `mpv-dev-x86_64` package), will make Plex work with RTX Super Resolution.

For RTX HDR you need at least 0.40.0. But that causes some issues for SDR files when HDR is enabled for me in Plex.
